network/hive.go

```diff
@@ -138,14 +137,17 @@ func (h *Hive) connect() {
 			continue
 		}
 
-		log.Trace(fmt.Sprintf("%08x hive connect() suggested %08x", h.BaseAddr()[:4], addr.Address()[:4]))
-		under, err := enode.ParseV4(string(addr.Under()))
-		if err != nil {
-			log.Warn(fmt.Sprintf("%08x unable to connect to bee %08x: invalid node URL: %v", h.BaseAddr()[:4], addr.Address()[:4], err))
-			continue
+		if h.Discovery {
```

Comment:
Note that if `h.addPeer` is not under the `if h.Discovery` clause, we were still making connections due to `hive` even if `h.Discovery` was set to `false`...

Reply:
lets remove checking of Discovery here, simply the connect loop should not be started